test/jb_splits_store/set.test.js

```diff
@@ -0,0 +1,278 @@
+
+    // Compare every currently stored splits to the one we've just sent
+    for (let [idx, split] of splitsStored) {
+      for (let splitKey of Object.keys(split)) {
+        expect(split.splitKey).to.equal(splits[idx].splitKey);
+      }
+    }
+  })
+
+  it('New splits without including a preexisting locked one: revert', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits } = await setup();
+
+    // Set one locked split
+    splits[1].lockedUntil = Date.now() + ONE_DAY;
+    let tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    );
+
+    // Try to set 4 new ones, with a new beneficiary for each, without the previous locked one
+    let newBeneficiary = addrs[5].address;
+    let newSplits = makeSplits(newBeneficiary);
+
+    await expect(
+      jbSplitsStore.connect(projectOwner).set(
+        PROJECT_ID,
+        DOMAIN,
+        GROUP,
+        newSplits
+      )
+    ).to.be.revertedWith('0x0f: SOME_LOCKED');
+  })
+
+  it('New splits with extension of a preexisting locked one', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits } = await setup();
+
+    // Set one locked split
+    splits[1].lockedUntil = Date.now() + ONE_DAY;
+    let tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    );
+
+    // Try to set new ones, with lock extension
+    let newSplits = makeSplits(addrs[5].address);
+
+    // New lockedUntil = old lockedUntil + 3600 sec
+    let newLockedTimestamp = Date.now() + 2 * (ONE_DAY);
+    newSplits[1].lockedUntil = newLockedTimestamp;
+    newSplits[1].beneficiary = addrs[0].address;
+
+    tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      newSplits
+    );
+
+    // Get the splits[] curently stored
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    expect(splitsStored[1].lockedUntil).to.equal(newLockedTimestamp);
+  })
+
+  it('One split with a percent at 0: revert', async function () {
+    const { projectOwner, jbSplitsStore, splits } = await setup();
+
+    // Set one 0% split
+    splits[1].percent = 0;
+
+    await expect(
+      jbSplitsStore.connect(projectOwner).set(
+        PROJECT_ID,
+        DOMAIN,
+        GROUP,
+        splits)
+    ).to.be.revertedWith('0x10: BAD_SPLIT_PERCENT');
+  })
+
+  it('Allocator and beneficiary both equal to address(0): revert', async function () {
+    const { projectOwner, jbSplitsStore, splits } = await setup();
+
+    // Set both allocator and beneficiary as address(0)
+    splits[1].beneficiary = ethers.constants.AddressZero;
+    splits[1].allocator = ethers.constants.AddressZero;
+
+    await expect(
+      jbSplitsStore.connect(projectOwner).set(
+        PROJECT_ID,
+        DOMAIN,
+        GROUP,
+        splits)
+    ).to.be.revertedWith('0x11: ZERO_ADDRESS');
+  })
```

Comment:
There should be separate tests for the beneficiary and for the allocator.

Reply:
Default for other tests is beneficiary=a non-zero address and allocator=address(0)
I will modify the first happy path test then add one with beneficiary=address(0) and allocator=non-zero